packages/core/http/core-http-server/src/router/utils.ts

```diff
@@ -37,3 +38,17 @@ export function getRequestValidation<P, Q, B>(
   if (typeof value === 'function') value = value();
   return isFullValidatorContainer(value) ? value.request : value;
 }
+
+/**
+ * Extracts {@link RouteValidatorFullConfigRequest} from the validation container.
+ * This utility is intended to be used by code introspecting router validation configuration.
+ * @public
+ */
+export function getResponseValidation(
+  value:
+    | RouteValidator<unknown, unknown, unknown>
+    | (() => RouteValidator<unknown, unknown, unknown>)
+): undefined | RouteValidatorFullConfigResponse {
+  if (typeof value === 'function') value = value();
```

Comment:
That's a good point... I assumed this lazy loading case would be uncommon but it could still result in performance degradation if the API gets hit a lot.

FWIW I believe the heaviest part of joi is calls to validate, but I agree might as well not introduce unnecessary new operations

Reply:
> the heaviest part of joi is calls to validate

Surprisingly, the schema creation itself is quite expensive (even if less than the actual validation, that's correct).

But it's not a blocker or anything